src/pool.rs

```diff
@@ -136,10 +139,14 @@ pub struct ConnectionPool {
     /// The server information (K messages) have to be passed to the
     /// clients on startup. We pre-connect to all shards and replicas
     /// on pool creation and save the K messages here.
-    server_info: BytesMut,
+    server_info: Arc<RwLock<BytesMut>>,
```

Comment:
This gets only read when a new client connects, so it should not have an contention on it.